pkg/ingester/ingester.go

```diff
@@ -1625,6 +1627,8 @@ func (i *Ingester) QueryStream(req *client.QueryRequest, stream client.Ingester_
 	spanlog, ctx := spanlogger.NewWithLogger(stream.Context(), i.logger, "Ingester.QueryStream")
 	defer spanlog.Finish()
 
+	deadline, deadlineSet := ctx.Deadline()
```

Comment:
[nit] I would move it before the call to `dumpContextError()`

Reply:
I moved it into `dumpContextError` since I don't see any downside to doing so (and the code gets simpler).